frontend/app-development/layout/SettingsModalButton/SettingsModal/components/Tabs/AccessControlTab/AccessControlTab.tsx

```diff
@@ -49,12 +79,38 @@ export const AccessControlTab = ({ org, app }: AccessControlTabProps): ReactNode
     updateAppMetadataMutation({ ...appMetadata, partyTypesAllowed: newPartyTypesAllowed });
   };
 
-  const displayCheckboxes = () => {
-    return getPartyTypesAllowedOptions().map((option) => (
-      <Checkbox value={option.value} key={option.value} size='small'>
-        {t(option.label)}
-      </Checkbox>
-    ));
+  const isAnyCheckboxChecked = checkedCheckboxes.length > 0;
+
+  const areAllCheckboxesChecked = checkedCheckboxes.length === getPartyTypesAllowedOptions().length;
+
+  const tableHeaderCheckboxState = isAnyCheckboxChecked
+    ? areAllCheckboxesChecked
+      ? CheckboxState.Checked
+      : CheckboxState.Indeterminate
+    : CheckboxState.Unchecked;
+
+  const handleTableHeaderCheckboxChange = () => {
+    if (tableHeaderCheckboxState === CheckboxState.Checked) {
+      setCheckedCheckboxes([]);
+    } else {
+      const updatedCheckboxes = getPartyTypesAllowedOptions().map((option) => option.value);
+      setCheckedCheckboxes(updatedCheckboxes);
+    }
+  };
+  const renderModal = () => {
+    return (
+      <Modal ref={modalRef}>
+        <Modal.Header />
+        <Modal.Content>
+          {t('settings_modal.access_control_tab_option_choose_type_modal_message')}
+        </Modal.Content>
+        <Modal.Footer>
+          <Button variant='secondary' onClick={() => modalRef.current?.close()}>
+            {t('general.close')}
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
```

Comment:
The modal has two ways to close it, is that necessary? 
![Skjermbilde 2024-03-18 kl  15 39 00](https://github.com/Altinn/altinn-studio/assets/71079896/90cd8bdb-5676-48ce-9080-e8a2a347abdf)

Reply:
I will remove the x 😊

Reply:
Kan vi endre meldingen her? "Du kan ikke ta bort det siste valget. Minst en part må være valgt."

Reply:
fixed